llvm/lib/Transforms/IPO/DeadArgumentElimination.cpp

```diff
@@ -714,6 +758,83 @@ void DeadArgumentEliminationPass::PropagateLiveness(const RetOrArg &RA) {
   Uses.erase(Begin, I);
 }
 
+// Update kernel arguments table inside the integration header.
+// For example:
+//   static constexpr const bool param_omit_table[] = {
+//     // OMIT_TABLE_BEGIN
+//     // kernel_name_1
+//     false, false,     // <= update to true if the argument is dead
+//     // kernel_name_2
+//     false, false,
+//     // OMIT_TABLE_END
+//   };
+//   TODO: batch changes to multiple SYCL kernels and do one bulk update.
+constexpr StringLiteral OMIT_TABLE_BEGIN("// OMIT_TABLE_BEGIN");
+constexpr StringLiteral OMIT_TABLE_END("// OMIT_TABLE_END");
+static void updateIntegrationHeader(StringRef SyclKernelName,
+                                    const ArrayRef<bool> &ArgAlive) {
+  // TODO: Enable asserts once the driver will provide the option.
```

Comment:
I'm not sure I get this comment. I see uses of assert function in this file, so it should be fine to use.

Reply:
I did not enable assert here since it will break many things because the driver does not provide this option right now.

Reply:
Discussed offline with Denis: the assert is supposed to check that pass is ran if and only if `IntegrationHeaderFileName` option is set, which is currently not true. Especially if the pass manager is populated by non-clang tool (e.g. opt).